src/fp/fp-capture.tsx

```diff
@@ -0,0 +1,85 @@
+import { Component, h, createRef, RefObject } from 'preact';
+import { createPortal } from 'preact/compat';
+import { ClientMessageType, HostMessage, HostMessageType, FingerprintComponents } from './types';
+import { FP_CAPTURE_URL } from '../core';
+
+interface FpCaptureProps {
+    requestCapture: boolean;
+    onCaptured: (components: FingerprintComponents) => void;
+}
+
+interface FpCaptureState {
+    isReady: boolean;
+    captureRequested: boolean;
+}
+
+export default class FpCapture extends Component<
+    FpCaptureProps,
+    FpCaptureState
+    > {
+
+    private iFrameRef: RefObject<HTMLIFrameElement>;
+
+    constructor(props: FpCaptureProps) {
+        super(props);
+
+        this.iFrameRef = createRef<HTMLIFrameElement>();
+
+        this.state = {
+            isReady: false,
+            captureRequested: this.props.requestCapture
+        };
+    }
+
+    componentDidMount() {
+        window.addEventListener('message', this.onMessage);
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('message', this.onMessage);
+    }
+
+    componentDidUpdate(prevProps: FpCaptureProps) {
+        if (this.props.requestCapture === this.state.captureRequested) {
+            return;
+        }
+
+        const captureRequested = this.props.requestCapture && !this.state.captureRequested;
+
+        this.setState({ captureRequested }, this.requestFingerprint);
+    }
+
+    private onMessage = (e: MessageEvent) => {
+        const message = e.data;
```

Comment:
Worth guarding if the message comes from the origin of our PD runtime?

Reply:
Yes, makes sense - done 👍